### PR TITLE
Pyt 1684 fastapi

### PR DESCRIPTION
**Pull request by danilito19**
Make initial fastapi app work with make and add auto-generated routes.

fast

### CODE REVIEW

**Comment by danilito19** [comment]
@lazorchakp  I'm pretty baffled by the failing pickle tests ... it only happens for fastapi, and it doesn't happen for me locally! Try/catching at `do_pickle_load*` or at the test itself seems wrong....

**Comment by lazorchakp** [comment]
I'm checking it out locally and will let you know what I find

File: src/vulnpy/fastapi/vulnerable.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+def get_trigger_name(name, trigger):
+    return "/{}/{}/".format(name, trigger)
+
+
+def gen_root_view(name):
+    view_name = get_root_name(name)
+
+    @router.get(view_name)
+    def _view():
+        return get_template("{}.html".format(name))
```
Comment (lazorchakp) [inline src/vulnpy/fastapi/vulnerable.py:25]: PYT-1705 can be resolved pretty quickly by returning an instance of `HTMLResponse`. So both here and in `_view` below, something like
```
return HTMLResponse(template)
```
the import looks like
```
from fastapi.responses import HTMLResponse
```

File: tests/requirements.txt [hunk 2]
```diff
@@ -1,2 +1,3 @@
 pytest-cov
 tox
+requests
```
Comment (lazorchakp) [inline tests/requirements.txt:3]: is this necessary? I can't find any uses of requests

File: apps/fastapi_app.py [hunk 3]
```diff
@@ -4,13 +4,16 @@
...
-
-    app.middleware("http")(ContrastMiddleware(app))
+# TODO: PYT-1701
+# if os.environ.get("VULNPY_USE_CONTRAST"):
+#     from contrast.fastapi import ContrastMiddleware
+#
+#     app.middleware("http")(ContrastMiddleware(app))
 
 
 @app.get("/")
```
Comment (lazorchakp) [inline apps/fastapi_app.py:19]: this is not hugely important (and I'm not sure if we're relying on this behavior somewhere else) but all other vulnpy example apps redirect `/` to `/vulnpy`

**Review by lazorchakp — CHANGES_REQUESTED** [review]
Requesting a few changes. I'm still looking into the test failures but I wanted to at least give you feedback on everything else.

It looks like something about fastapi (or maybe its test client) is removing `\n` from the querystring before it hits the view in the pickle tests. ie
```
flask (correct):
csubprocess\ncheck_output\n(S'ls'\ntR.

fastapi (incorrect)
csubprocesscheck_output(S'ls'tR.
```

File: tests/fastapi/test_vulnerable.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+
+@parametrize_triggers
+def test_trigger(client, view_name, trigger_name):
+    data = DATA[view_name]
+
+    if view_name == "unsafe_code_exec":
+        data = "'{}'".format(data)
+
+    response = client.get(
```
Comment (lazorchakp) [inline tests/fastapi/test_vulnerable.py:36]: I figured out the pickle error. This should be:
```
    response = client.get(
        "/vulnpy/{}/{}/".format(view_name, trigger_name), params={"user_input": data}
    )
```
Which causes the TestClient to correctly urlencode the query parameter

File: src/vulnpy/fastapi/vulnerable.py (hunk 1 above)
Comment (danilito19) [reply src/vulnpy/fastapi/vulnerable.py:25]: oooo

File: tests/requirements.txt (hunk 2 above)
Comment (danilito19) [reply tests/requirements.txt:3]: unfortunately yes... the fastapi / starlette testclient requires requests

Comment (lazorchakp) [reply tests/requirements.txt:3]: oh damn

**Comment by danilito19** [comment]
Peter thanks for the good feedback! Updated description with new gif showing html rendering + redirect

**Review by lazorchakp — APPROVED** [review]

**Pull request merged**